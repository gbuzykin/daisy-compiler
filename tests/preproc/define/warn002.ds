#define MACRO 0
#define MACRO 1
const a = MACRO;