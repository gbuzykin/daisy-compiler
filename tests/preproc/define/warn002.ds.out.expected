./preproc/define/warn002.ds:2:9: warning: macro `MACRO` redefinition
 2 | #define MACRO 1
   |         ^~~~~
./preproc/define/warn002.ds:3:1: debug: token
 3 | const a = MACRO;
   | ^~~~~
./preproc/define/warn002.ds:3:7: debug: id: a
 3 | const a = MACRO;
   |       ^
./preproc/define/warn002.ds:3:9: debug: token
 3 | const a = MACRO;
   |         ^
./preproc/define/warn002.ds:3:11: debug: integer number: 1
 3 | const a = MACRO;
   |           ^~~~~
./preproc/define/warn002.ds:2:15: note: expanded from macro `MACRO`
 2 | #define MACRO 1
   |               ^
./preproc/define/warn002.ds:3:7: debug: defining constant `a`
 3 | const a = MACRO;
   |       ^
./preproc/define/warn002.ds:3:16: debug: token
 3 | const a = MACRO;
   |                ^
./preproc/define/warn002.ds: info: warnings 1, errors 0
